language/src/main/scala/com/reactific/riddl/language/ast/Actions.scala

```diff
@@ -9,7 +9,8 @@ package com.reactific.riddl.language.ast
 /** This trait defines all the Actions that can be invoked from an Example and
   * classified by the kind of definition to which they are applicable
   */
-trait Actions extends Definitions {
+trait Actions {
+  this: Definitions with Expressions with Types with Options with AbstractDefinitions =>
 
```

Comment:
This makes Actions not derive from Definitions which means the Actions subclasses won’t either. But, the tests pass so maybe the self type is sufficient?  Not sure how much I care. This will all likely get rewritten someday

Reply:
The only concrete type that I see which extends these traits is `AST`. The rest of riddl only addresses anything in the ast package via `AST`. The self type requires any concrete implementation of Actions to also implement the traits it depends on, so everything works out.

To be clear, my fix is certainly a scala type-system trick to flatten the inheritance graph and make scalac happy. But the traits only exist as an trick to define `AST` across multiple files anyways. We could just drop the AST object instead and import the `ast` package. That would also avoid the scalac bug, but that's a much larger change overall. 

I don't see any way forward on getting docs to work without flattening the heirarchy somehow or waiting for the bug to be fixed. With current scala versions, the current inheritance graph is just too complicated for scaladoc to deal with :/

Reply:
If you are only targetting Scala 3 we could just turn the traits insto objects, do normal imports within the ast package, then export all the objects from `AST` to retain the same API and remove the inheritance entirely.

Reply:
Thanks for the thoughts. There are several objectives I'm trying to achieve here. Originally everything in ast/*.scala was in one file, AST.scala.  That got unwieldy in 2.13,  so I used traits to break it apart. This approach is confounded by the desire to declare a large number of sealed traits which must have all their case classes in the same compilation unit.  I've not really used any Scala 3 level design yet, just ported the code (a few weeks ago).  Probably the right thing to do is to place all these things at file scope and import "ast/*" instead of "AST.*". Just ... too much for me to handle with current workload.